src/daemon/mod.rs

```diff
@@ -7,48 +7,28 @@ mod sigfetcher;
 mod threadmessages;
 mod utils;
 
-use crate::{
+use crate::assume_ok;
+use crate::daemon::{
     bitcoind::{bitcoind_main_loop, start_bitcoind},
     control::RpcUtils,
     database::actions::setup_db,
     jsonrpc::{
         server::{rpcserver_loop, rpcserver_setup},
         UserRole,
     },
-    revaultd::RevaultD,
     sigfetcher::signature_fetcher_loop,
 };
-use common::{assume_ok, config::Config};
-use revault_net::sodiumoxide;
-use revault_tx::bitcoin::hashes::hex::ToHex;
+
+pub use crate::daemon::revaultd::RevaultD;
 
 use std::{
-    env,
     io::{self, Write},
-    panic,
-    path::PathBuf,
-    process,
+    panic, process,
     sync::{mpsc, Arc, RwLock},
     thread, time,
 };
 
-use daemonize_simple::Daemonize;
-
-fn parse_args(args: Vec<String>) -> Option<PathBuf> {
-    if args.len() == 1 {
-        return None;
-    }
-
-    if args.len() != 3 {
-        eprintln!("Unknown arguments '{:?}'.", args);
-        eprintln!("Only '--conf <configuration file path>' is supported.");
-        process::exit(1);
-    }
-
-    Some(PathBuf::from(args[2].to_owned()))
-}
-
-fn daemon_main(mut revaultd: RevaultD) {
+pub fn daemon_main(mut revaultd: RevaultD) {
```

Comment:
Is there a rationale of why you're leaving some methods here? Isn't it better to move everything to `bin`?

Reply:
Yes these methods will be required by the GUI to run the daemon in a thread

Reply:
Ah right 👍🏻